### PR TITLE
Added Powershell script to auto-generate the merged headers for a client SDK

### PR DESCRIPTION
**Pull request by LarryOsterman**
Created a powershell script to automate the process of creating the merged headers for a client SDK.

Usage:
```pwsh
 .\eng\scripts\Generate-Unified-Header.ps1 -PackageName azure-security-keyvault-secrets -ServiceName keyvault  -OutputHeaderName "keyvault/secrets.hpp" -PackageDisplayName "Key Vault Secrets"
```

Command line switches:
- ServiceName - name of the service directory.
- PackageName - name of the package directory under the service directory.
- OutputHeaderName - name of the output header under `<root>\sdk\<ServiceName>\<PackageName>\inc\azure` directory.
- PackageDisplayName - If provided, the human readable name of the package, used for a header comment.
- OutputFile - the name of the output file - if provided, overrides OutputHeaderName.

Eventually I'd like to run this for all of the services and packages and then integrate it into the CI pipeline to ensure that the unified headers are kept in sync with the source tree.

# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

File: sdk/core/azure-core-tracing-opentelemetry/inc/azure/core/tracing.hpp [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#pragma once
+
+#include "azure/core/tracing/opentelemetry/dll_import_export.hpp"
+#include "azure/core/tracing/opentelemetry/internal/apiview.hpp"
```
Comment (antkmsft) [inline sdk/core/azure-core-tracing-opentelemetry/inc/azure/core/tracing.hpp:6]: `/internal/` shouldn't be included.

File: sdk/core/azure-core-tracing-opentelemetry/inc/azure/core/tracing.hpp [hunk 2]
```diff
@@ -0,0 +1,8 @@
+// Copyright (c) Microsoft Corporation.
```
Comment (antkmsft) [inline sdk/core/azure-core-tracing-opentelemetry/inc/azure/core/tracing.hpp:1]: The header should be named `opentelemetry.hpp`, and it should be in `azure/core/tracing/`

File: sdk/core/azure-core-tracing-opentelemetry/inc/azure/core/tracing.hpp (hunk 1 above)
Comment (LarryOsterman) [reply sdk/core/azure-core-tracing-opentelemetry/inc/azure/core/tracing.hpp:6]: DOne. This file was created before I added the "remove internal" filter.

**Review by antkmsft — APPROVED** [review]

File: eng/scripts/Generate-Unified-Header.ps1 [hunk 3]
```diff
@@ -0,0 +1,63 @@
+param(
```
Comment (ahsonkhan) [inline eng/scripts/Generate-Unified-Header.ps1:1]: We are missing copyright header comment here.

File: eng/scripts/Generate-Unified-Header.ps1 [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  if ($file -NE $OutputFile)
+  {
+    if ($file -like "*_transport.hpp")
+    {
+      continue;
+    }
+    if ($file -like "*keyvault_keys.hpp" -and $PackageName -EQ "azure-security-keyvault-keys")
+    {
+      continue;
+    }
```
Comment (ahsonkhan) [inline eng/scripts/Generate-Unified-Header.ps1:56]: Why do we have these hardcoded `keyvault_keys` and `_transport` specific strings in this general-purpose script? Maybe leftover from testing and should be removed.

Comment (LarryOsterman) [reply eng/scripts/Generate-Unified-Header.ps1:56]: These are there to ensure that we generate correct headers.

For instance, in core, we explicitly remove the HTTP transports from the combined header.

keyvault_keys.hpp is another special case - it is a legacy header which does nothing but #include "keys.hpp". And it cannot go into the unified header.

Comment (ahsonkhan) [reply eng/scripts/Generate-Unified-Header.ps1:56]: A comment explaining these hard-coded edge cases would be a good idea. It also helps when we need to add more exceptions in the future.

**Pull request merged**